### ISSUE TITLE
HTML Form Fields do not convert properly into PDF

### ISSUE THREAD

**Issue by DunhamGitHub** [wkhtmltopdf/wkhtmltopdf#4275]
**wkhtmltopdf version(s) affected**: 0.12.5

**OS information**
Centos 7

**Description**
conver simple html file with '<input type="text" value="" />' to pdf -> form field does not allow editing

**How to reproduce**
Html file:
```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>test page edit 2</title>
</head>
<body>
<form>
<input type="text" name="test" id="inputtext" value="" />
</form>
</body>
</html>
```

`wkhtmltopdf file.html output.pdf`

Not editable ->
![image](https://user-images.githubusercontent.com/6884752/53885570-61789600-401e-11e9-97a8-4c8dcb9a9cc8.png)

**Expected behavior**
Form fields should be editable - here is how the Adobe Acrobat Pro version looks (when I manually add form fields)
![image](https://user-images.githubusercontent.com/6884752/53885518-3db55000-401e-11e9-970e-1ae9b2c1682d.png)

**Comment by shaunkatona** [comment]
@DunhamGitHub, have you tried `wkhtmltopdf --enable-forms file.html output.pdf`?

**Comment by DunhamGitHub** [comment]
Yes I did (mistake in my ticket here).
Just tried again: wkhtmltopdf --enable-forms convertme.html form2.pdf
Ill attach the file (form field NOT editable)
I have seen other tickets with this issue, seems noone cares too much, c'est la vie - will probably have to use iText.
[form2.pdf](https://github.com/wkhtmltopdf/wkhtmltopdf/files/2960268/form2.pdf)

**Comment by DunhamGitHub** [comment]
Seems the form is editable in Chrome -> Yet not in Acrobat, which is not good enough, you cannot tell a client / user to use a different tool just for certain PDFs

**Comment by shaunkatona** [comment]
Yeah - it works in Chrome, and in Mac's Preview app.  I don't have Acrobat to test with, but do you have the most up to date version of Acrobat?  If that's the outlier, that seems weird...

**Comment by warkior** [comment]
I'm experiencing the same problem. I put a plain html file with input forms on desktop and run `wkhtmltopdf --enable-forms 'Desktop/document.html' 'Desktop/document.pdf'` but the fields on the generated PDF document are not editable.

    Version: wkhtmltopdf 0.12.5 (with patched qt)
    Command: wkhtmltopdf --enable-forms 'Desktop/document.html' 'Desktop/document.pdf'
    OS: Mac OS 10.14.3

I have confirmed that input fields appear to work in Chrome but textareas and checkboxes do not. None of the fields appear to work in Adobe Acrobat.

**Comment by DunhamGitHub** [comment]
As of this writing Acrobat's most recent version is Version 2019.010.20098. This is what I have - tested it on 2 machines, input not editable.
I would say the lion share of users / end-clients use Acrobat Reader (because they have a Windows PC)

**Comment by aldefouw** [comment]
Seeing the same behavior in 2020.  Guessing no fix has been found ...

**Comment by entrity** [comment]
Even in Chrome, if you set a `value=...` attribute on the form fields, they don't display correctly: if you edit the field, the original `value` stays visible behind the new text:

![image](https://user-images.githubusercontent.com/950824/81991196-8179f100-95f5-11ea-8c1d-56746380cf41.png)

...and checkboxes are not settable/clearable.

**Comment by anasirdev** [comment]
what about if the form input field type is file?? how can you generate pdf from the uploaded document??
<input type="file">

**Comment by axlle-com** [comment]
Seeing the same behavior in 2021. Guessing no fix has been found ...

**Comment by samalone** [comment]
According to [this answer on Stack Overflow](https://stackoverflow.com/a/10666574), Adobe has specifically engineered Acrobat to distinguish between Acrobat-generated forms and forms created by third-party software. Without reverse-engineering this format, it is not possible to create Acrobat-compatible forms.

**Comment by DunhamGitHub** [comment]
> According to [this answer on Stack Overflow](https://stackoverflow.com/a/10666574), Adobe has specifically engineered Acrobat to distinguish between Acrobat-generated forms and forms created by third-party software. Without reverse-engineering this format, it is not possible to create Acrobat-compatible forms.

This _stackoverflow article_ only states that Adobe Reader cannot save! filled forms **(but it can fill+save).** 
The issue here is that the form created with _wkhtml2pdf_ is not even editable.

<img width="1737" alt="Screen Shot 2022-05-25 at 10 52 10" src="https://user-images.githubusercontent.com/6884752/170222860-1b2e0379-70bb-4d99-8e91-34b2ed2f7a53.png">
